internal/process/upgrade_cluster/initialisation.go

```diff
@@ -96,10 +96,13 @@ func (s *InitialisationStep) Run(operation internal.UpgradeClusterOperation, log
 			}
 		}
 
-		op, delay, _ := s.operationManager.UpdateOperation(operation, func(op *internal.UpgradeClusterOperation) {
+		op, delay, err := s.operationManager.UpdateOperation(operation, func(op *internal.UpgradeClusterOperation) {
 			op.ProvisioningParameters.ErsContext = internal.InheritMissingERSContext(op.ProvisioningParameters.ErsContext, lastOp.ProvisioningParameters.ErsContext)
 			op.State = domain.InProgress
 		}, log)
+		if err != nil {
+			log.Infof("error during update operation initialisation: %s", err)
```

Comment:
Is this really needed? The message was already logged as error in UpdateOperation.

Reply:
```go
func (om *OperationManager) UpdateOperation(operation internal.Operation, update func(operation *internal.Operation), log logrus.FieldLogger) (internal.Operation, time.Duration, error) {
	update(&operation)
	op, err := om.storage.UpdateOperation(operation)
	switch {
	case dberr.IsConflict(err):
		{
			op, err = om.storage.GetOperationByID(operation.ID)
			if err != nil {
				log.Errorf("while getting operation: %v", err)
				return operation, 1 * time.Minute, err
			}
			op.Merge(&operation)
			update(op)
			op, err = om.storage.UpdateOperation(*op)
			if err != nil {
				log.Errorf("while updating operation after conflict: %v", err)
				return operation, 1 * time.Minute, err
			}
		}
	case err != nil:
		log.Errorf("while updating operation: %v", err)
		return operation, 1 * time.Minute, err
	}

	return *op, 0, nil
}
```